### ISSUE TITLE
Is this only for WordPress?

### ISSUE THREAD

**Issue by IonicaBizau** [pojome/elementor#1096]
In the installation steps it says that we have to install WordPress. Is it possible to run Elementor without WordPress?

It looks good, but I don't want to depend on WordPress. I just want to have a standalone version which can output static HTML files.

Is that possible?

**Comment by arielk** [comment]
No, Elementor is only for WordPress.

**Comment by IonicaBizau** [comment]
What would be the right way to make it framework agnostic?

**Comment by arielk** [comment]
Good question, actually I don't see a way to implement it, this application is built for WordPress. But it's open source, if you're a developer you can try :)

**Comment by iamntz** [comment]
How I would do it:

1. Create an interface - let's say `Cms` - that requires various methods to be implemented;
1. Extract all WP related functions into a class (or multiple classes, whatever) that implements that interface. E.g. `get_post_meta` would be `Cms\WP->get_meta`, `wp_verify_nonce` would be `Cms\WP->verify_nonce` and so on.
1. Pass these implementations around and instead of `get_post_meta` you would use `$this->cms->get_post_meta` or something similar.
1. Implement the `Cms` interface for every CMS you want, so you'll have `Cms\Drupal->get_meta` as a wrapper for whatever Drupal uses and so on.

So, long story short:

1. it is doable? Surely it is!
1. it is easy? Let's say that it's not really a weekend project (unless you're super excited and motivated about it!)
1. It is worth it? Hmmm... hard to say, although it would be super nice to have something like Elementor for.. let's say Jekyll, Middleman or whatever

**Comment by IonicaBizau** [comment]
@iamntz I agree! I'd actually go in this direction: separate the UI into a JS library kind of thing, and create adapters for the server side (e.g. WordPress). In my case I'd like to do it with Node.js because I'm not really a PHP guy. What I do like is the simplicity of this editor. :art: 

PS: Greetings from Bihor! :joy:

**Comment by TolstoyDotCom** [comment]
There are over 400 instances of things starting with "wp_" in the code. Some of those wouldn't need to be replaced. Something like `new \WP_Error( 'scheme_class_name_not_exists' )` could be fairly easily provided if it doesn't exist or replaced with a generic class. More daunting would be the uses of add_action and add_filter. For instance, there's `add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_scripts' ] )` and that method calls `wp_register_script`. There are different ways to add scripts in Drupal, but one possible way to do that would be to replace the add_action with `drupal_add_js`. Other uses of add_action would probably be more problematic.
